commons-vfs2/src/main/java/org/apache/commons/vfs2/util/MonitorOutputStream.java

```diff
@@ -121,7 +121,9 @@ public void close() throws IOException {
      */
     @Override
     public synchronized void flush() throws IOException {
-        assertOpen();
```

Comment:
Why was the open check completely removed?

Reply:
I think it can be fixed this way because:

1. we will invoke **flush** when **close**.

2. after invoke **close**, every operations that will try to add something to the buffer would cause a  FileSystemException

3. then if invoke **flush** after close, there shouold actually nothing to flush

Reply:
So I don't think we really need to forbidden **flush** after **close**.
Just do nothing and return should be fine IMO